client/apis/volumegroupsnapshot/v1alpha1/types.go

```diff
@@ -0,0 +1,284 @@
+	Status VolumeGroupSnapshotStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// VolumeGroupSnapshotList contains a list of VolumeGroupSnapshot
+type VolumeGroupSnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VolumeGroupSnapshot `json:"items"`
+}
+
+//+genclient
+//+genclient:nonNamespaced
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeGroupSnapshotClass is the Schema for the volumegroupsnapshotclass API
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,shortName=vsgclass;vsgclasses
+type VolumeGroupSnapshotClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Driver is the name of the storage driver expected to handle this VolumeGroupSnapshotClass.
+	// Required.
+	Driver string `json:"driver" protobuf:"bytes,2,opt,name=driver"`
+
+	// Parameters is a key-value map with storage driver specific parameters for
+	// creating group snapshots.
+	// These values are opaque to the system and are passed directly
+	// to the driver.
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters"`
+
+	// DeletionPolicy determines whether a VolumeGroupSnapshotContent created
+	// through the VolumeGroupSnapshotClass should be deleted when its bound
+	// VolumeGroupSnapshot is deleted.
+	// Supported values are "Retain" and "Delete".
+	// Required.
+	DeletionPolicy *snapshotv1.DeletionPolicy `json:"deletionPolicy" protobuf:"bytes,4,opt,name=deletionPolicy"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeGroupSnapshotClassList is a collection of VolumeGroupSnapshotClasses.
+// +kubebuilder:object:root=true
+type VolumeGroupSnapshotClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Items is the list of VolumeGroupSnapshotClass
+	Items []VolumeGroupSnapshotClass `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeGroupSnapshotContent represents the actual "on-disk" group snapshot object
+// in the underlying storage system
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,shortName=vsc;vgscs
+// +kubebuilder:subresource:status
+type VolumeGroupSnapshotContent struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   VolumeGroupSnapshotContentSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	Status VolumeGroupSnapshotContentStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
```

Comment:
Status *VolumeGroupSnapshotContentStatus
// status represents the current information of a snapshot.
	// +optional